src/lib/pipeline.py

```diff
@@ -370,31 +404,35 @@ def run(
 
         # Get all the pipeline outputs
         # This operation is parallelized but output order is preserved
-        map_iter = [
-            # Make a copy of the auxiliary table to prevent modifying it for everyone, but this way
-            # we allow for local modification (which might be wanted for optimization purposes)
-            (data_source, cache, {name: df.copy() for name, df in aux.items()})
-            for data_source in self.data_sources
-        ]
+
+        # Make a copy of the auxiliary table to prevent modifying it for everyone, but this way
+        # we allow for local modification (which might be wanted for optimization purposes)
+        aux_copy = {name: df.copy() for name, df in aux.items()}
+
+        # Create a function to be used during mapping. The nestedness is an unfortunate outcome of
+        # the multiprocessing module's limitations when dealing with lambda functions, coupled with
+        # the "sandboxing" we implement to ensure resiliency.
+        run_func = partial(DataPipeline._run_wrapper, output_folder, cache, aux_copy)
```

Comment:
More specific naming might be beneficial here. What does `run_func` do?

Reply:
I think it's pretty clear based on the code, since it's a single line: it creates a partial function out of `_run_wrapper`.